runtime/src/currency.rs

```
@@ -189,3 +199,122 @@ impl<AccountId> ReservableCurrency<AccountId> for CurrencyWaiver {
+		0
+	}
+	fn can_deposit(_: &AccountId, _: Self::Balance, _: Provenance) -> DepositConsequence {
+		DepositConsequence::Success
+	}
+	fn can_withdraw(_: &AccountId, _: Self::Balance) -> WithdrawConsequence<u128> {
+		WithdrawConsequence::Success
+	}
+}
+
+impl<AccountId: Eq> FungibleMutate<AccountId> for CurrencyWaiver {
+	fn mint_into(_: &AccountId, _: u128) -> Result<u128, DispatchError> {
+		Ok(0)
+	}
+	fn burn_from(
+		_: &AccountId,
+		_: u128,
+		_: Preservation,
+		_: Precision,
+		_: Fortitude,
+	) -> Result<u128, DispatchError> {
+		Ok(0)
+	}
+	fn transfer(
+		_: &AccountId,
+		_: &AccountId,
+		_: u128,
+		_: Preservation,
+	) -> Result<u128, DispatchError> {
+		Ok(0)
+	}
+	fn set_balance(_: &AccountId, _: u128) -> u128 {
+		0
+	}
+}
+
+impl<AccountId> FungibleInspectHold<AccountId> for CurrencyWaiver {
+	type Reason = HoldReason;
+	fn balance_on_hold(_: &Self::Reason, _: &AccountId) -> u128 {
+		0
+	}
+	fn reducible_total_balance_on_hold(_: &AccountId, _: Fortitude) -> u128 {
+		0
+	}
+	fn can_hold(_: &Self::Reason, _: &AccountId, _: u128) -> bool {
+		true
+	}
+	fn hold_available(_: &Self::Reason, _: &AccountId) -> bool {
+		true
+	}
+	fn total_balance_on_hold(_: &AccountId) -> Self::Balance {
+		0
+	}
+}
+
+impl<AccountId> FungibleMutateHold<AccountId> for CurrencyWaiver {
+	fn hold(_: &HoldReason, _: &AccountId, _: u128) -> Result<(), DispatchError> {
+		Ok(())
+	}
+	fn release(
+		_: &HoldReason,
+		_: &AccountId,
+		_: u128,
+		_: Precision,
+	) -> Result<u128, DispatchError> {
+		Ok(0)
+	}
+}
+
+impl<AccountId> FungibleUnbalanced<AccountId> for CurrencyWaiver {
+	fn handle_dust(_: frame_support::traits::fungible::Dust<AccountId, Self>) {
+		// Do nothing
+	}
+	fn write_balance(
+		_: &AccountId,
+		_: <Self as frame_support::traits::fungible::Inspect<AccountId>>::Balance,
+	) -> Result<
+		core::option::Option<
+			<Self as frame_support::traits::fungible::Inspect<AccountId>>::Balance,
+		>,
+		sp_runtime::DispatchError,
+	> {
+		todo!()
```

better to return an error than panic?